### PR TITLE
Recreate outputFileName completer, handle process launch errors.

### PR DESCRIPTION
**Pull request by aam**
This PR fixes few things around incremental kernel generator:
- New Future Completer has to be created on every `recompile` call. Previously the old one being reused for all successive recompilation requests.
- To prevent confusion when to make or skip `accept` call following incremental kernel generator compile request, incremental kernel generator will be asked to do compilation even if there were no changes made to the source.
- Added `catchError` for process invocations.

### CODE REVIEW

**Comment by Hixie** [comment]
cc @cbracken or @tvolkert

**Comment by aam** [comment]
PTAL, when you have a chance

File: packages/flutter_tools/lib/src/devfs.dart [hunk 1]
```diff
@@ -429,20 +429,26 @@ class DevFS {
...
+      // We run generator even if [dirtyEntries] was empty because we want
+      // to keep logic of accepting/rejecting generator's output simple:
+      // we must accept/reject generator's output after every [update] call.
+      // Incremental run with no changes is supposed to be fast (considering
+      // that it is initiated by user key press).
+      final List<String> invalidatedFiles = <String>[];
+      for (DevFSContent content in dirtyEntries.values)
+        if (content is DevFSFileContent)
+          invalidatedFiles.add(content.file.uri.toString());
+      printTrace('Compiling dart to kernel with ${invalidatedFiles.length} updated files');
```
Comment (a-siva) [inline packages/flutter_tools/lib/src/devfs.dart]: printTrace for debugging? can it be removed?

File: packages/flutter_tools/lib/src/devfs.dart [hunk 2]
```diff
@@ -429,20 +429,26 @@ class DevFS {
           assetPathsToEvict.add(archivePath);
       }
     });
+    if (generator != null) {
+      // We run generator even if [dirtyEntries] was empty because we want
+      // to keep logic of accepting/rejecting generator's output simple:
+      // we must accept/reject generator's output after every [update] call.
+      // Incremental run with no changes is supposed to be fast (considering
+      // that it is initiated by user key press).
+      final List<String> invalidatedFiles = <String>[];
```
Comment (a-siva) [inline packages/flutter_tools/lib/src/devfs.dart]: Should this be  final var invalidatedFiles = <String>[];

File: packages/flutter_tools/lib/src/run_hot.dart [hunk 3]
```diff
@@ -135,6 +135,10 @@ class HotRunner extends ResidentRunner {
 
     await refreshViews();
     for (FlutterDevice device in flutterDevices) {
+      // VM has to accept the kernel binary, there will be no reload report,
```
Comment (a-siva) [inline packages/flutter_tools/lib/src/run_hot.dart]: Should the comment be VM has accepted the kernel binary?

File: packages/flutter_tools/lib/src/run_hot.dart [hunk 4]
```diff
@@ -334,6 +338,10 @@ class HotRunner extends ResidentRunner {
       return new OperationResult(1, 'DevFS synchronization failed');
     // Check if the isolate is paused and resume it.
     for (FlutterDevice device in flutterDevices) {
+      // VM has to accept the kernel binary, there will be no reload report,
```
Comment (a-siva) [inline packages/flutter_tools/lib/src/run_hot.dart]: Ditto comment about "VM has accepted"

**Review by a-siva — COMMENTED** [review]
Some nit comments.

File: packages/flutter_tools/lib/src/devfs.dart (hunk 1 above)
Comment (aam) [reply packages/flutter_tools/lib/src/devfs.dart]: Not sure what are the guidelines for printTraces, but I think this one is helpful - between it and next `printTrace('Update files')` we can see for example how much time is spent in actual compilation.

File: packages/flutter_tools/lib/src/devfs.dart (hunk 2 above)
Comment (aam) [reply packages/flutter_tools/lib/src/devfs.dart]: I believe the Flutter style guide suggests to always have variable type explicitly spelled out.

File: packages/flutter_tools/lib/src/run_hot.dart (hunk 3 above)
Comment (aam) [reply packages/flutter_tools/lib/src/run_hot.dart]: Changed the wording along the lines you suggested here and below.

**Comment by aam** [comment]
PTAL when you have a chance

**Review by a-siva — APPROVED** [review]

**Pull request merged**